packages/table/src/Table.ts

```diff
@@ -0,0 +1,463 @@
+
+        if (!!this.selects) {
+            this.tableHeadCheckboxCell = document.createElement(
+                'sp-table-checkbox-cell'
+            );
+            const allSelected = this.selected.length === tableRows.length;
+
+            if (this.tableHeadCheckboxCell) {
+                this.tableHeadCheckboxCell.selectsSingle =
+                    this.selects === 'single';
+                this.tableHeadCheckboxCell.checked = allSelected;
+                this.tableHeadCheckboxCell.indeterminate =
+                    this.selected.length > 0 && !allSelected;
+            }
+
+            tableHead.insertAdjacentElement(
+                'afterbegin',
+                this.tableHeadCheckboxCell
+            );
+
+            tableRows.forEach((row) => {
+                const checkbox = document.createElement(
+                    'sp-table-checkbox-cell'
+                );
+                row.insertAdjacentElement('afterbegin', checkbox);
+                row.selected = this.selectedSet.has(row.value);
+                checkbox.checked = row.selected;
+            });
+        } else {
+            tableHead.querySelector('sp-table-checkbox-cell')?.remove();
+            tableRows.forEach((row) => {
+                row.checkboxCells[0]?.remove();
+                if (this.selected.length) {
+                    row.selected = this.selectedSet.has(row.value);
+                }
+            });
+        }
+    }
+
+    protected handleChange(event: Event): void {
+        // Contain the children's change events and convert it to a change event on the table
+        // if rowItem doesn't have a value, we assume it's in TableHead (naive!!!)
+        const { target } = event;
+        const { parentElement: rowItem } = target as HTMLElement & {
+            parentElement: TableRow;
+        };
+        if (!rowItem.value) {
+            const { checkbox } = target as TableCheckboxCell;
+            if (checkbox.checked || checkbox.indeterminate) {
+                this.selectAllRows();
+            } else {
+                this.deselectAllRows();
+            }
+        } else {
+            switch (this.selects) {
+                case 'single': {
+                    this.deselectAllRows();
+                    if (rowItem.selected) {
+                        this.selectedSet.add(rowItem.value);
+                        this.selected = [...this.selectedSet];
+                    }
+                    break;
+                    // if you have multiple selected, but selects='single',
+                    // we deselect all cells? hmm...
+                    // EXAMPLE: rows 1, 2, 4 are selected but selects is single.
+                    // we deselect 2 => 1 and 4 are STILL SELECTED. if you want
+                    // to select any item, it's going to deselect all the cells before
+                    // reselecting the cell you just clicked on, as seen above.
```

Comment:
Some of these comments could be useful, but I'm not sure they are currently in a useful state.